crates/metaslang/bindings/src/lib.rs

```diff
@@ -153,19 +306,61 @@ impl<'a, KT: KindTypes + 'static> Definition<'a, KT> {
     }
 
     pub fn get_definiens_cursor(&self) -> Option<Cursor<KT>> {
-        self.owner.definiens.get(&self.handle).cloned()
+        self.owner
+            .definitions_info
+            .get(&self.handle)
+            .and_then(|info| info.definiens.clone())
     }
 
     pub fn get_file(&self) -> Option<&'a str> {
         self.owner.stack_graph[self.handle]
             .file()
             .map(|file| self.owner.stack_graph[file].name())
     }
+
+    pub(crate) fn has_tag(&self, tag: Tag) -> bool {
+        self.owner
+            .definitions_info
+            .get(&self.handle)
+            .and_then(|info| info.tag)
+            .is_some_and(|definition_tag| definition_tag == tag)
+    }
+
+    pub(crate) fn resolve_parents(&self) -> Vec<Definition<'a, KT>> {
+        self.owner
+            .definitions_info
+            .get(&self.handle)
+            .map(|info| &info.parents)
+            .map(|handles| self.owner.resolve_handles(handles))
+            .unwrap_or_default()
```

Comment:
Not necessarily. The bindings context may be incomplete. Since this is used by the linearisation algorithm to resolve base contracts from the references in the inheritance list, it may be the case that the base contract is not yet loaded into the bindings context. Or it may be a typo perhaps.

Reply:
I had left a TODO [here](https://github.com/manastech/slang/blob/caf4bc0cc45d190f80d2b76da9e900946c31d5d6/crates/metaslang/bindings/src/lib.rs#L192), but it should be ok to remove.

TOL: do we need a way to collect non-fatal errors during the bindings resolution that the user can later check? This would be similar to the parse errors in the parser, and they would indicate to the user that something is not quite right and the results they are getting are incomplete at best.

Reply:
> do we need a way to collect non-fatal errors during the bindings resolution that the user can later check? 

Not for v1 ..